### PR TITLE
Maintenance listing

### LINKED ISSUE #340: Maintenance listing

### PR DESCRIPTION
**Pull request by SCedricThomas**
fix #340
- [x] Add a [changelog entry](https://changelog.scalingo.com/)

### CODE REVIEW

**Review by curzolapierre — DISMISSED** [review]
LGTM

File: maintenance.go [hunk 1]
```diff
@@ -30,3 +59,18 @@ func (c *Client) DatabaseUpdateMaintenanceWindow(ctx context.Context, app, addon
...
+type ListMaintenanceResponse struct {
+	Maintenance []Maintenance  `json:"maintenance"`
+	Meta        PaginationMeta `json:"meta"`
+}
+
+func (c *Client) DatabaseListMaintenance(ctx context.Context, app, addonID string, opts PaginationOpts) (ListMaintenanceResponse, error) {
+	var maintenanceRes ListMaintenanceResponse
+	err := c.DBAPI(app, addonID).SubresourceList(ctx, "databases", addonID, "maintenance", opts.ToMap(), &maintenanceRes)
+	if err != nil {
+		return ListMaintenanceResponse{}, errgo.Notef(err, "list database '%v' maintenance", addonID)
```
Comment (curzolapierre) [inline maintenance.go:73]:
```suggestion
		return ListMaintenanceResponse{}, errgo.Notef(err, "list database '%v' maintenances", addonID)
```

**Review by curzolapierre — COMMENTED** [review]
Just a typo, otherwise LGTM

File: maintenance.go (hunk 1 above)
Comment (curzolapierre) [reply maintenance.go:73]: As discussed, the maintenance noun doesn't take 's' in plural.

**Review by curzolapierre — APPROVED** [review]

File: maintenance.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (SCedricThomas) [reply maintenance.go:73]: We will tackle this issue as soon as possible

[the remark this answers is not in the document]
Comment (francois2metz) [reply maintenance.go:73]: Unrelated, but is the API available? It seems to returns 404 for now (trying to add the steampipe tables).

**Pull request merged**